backend/src/routes/api.routes.js

```diff
@@ -2,15 +2,22 @@ import express from 'express';
 import bodyParser from 'body-parser';
 const cors = require('cors');
 import { helloController } from '../controllers';
+import { UserController } from '../controllers/UserController';
+import { validateUser } from '../middlewares/validators/userValidator';
 import { sessionsController } from '../controllers';
 import { verifyToken } from '../services/sessionService'
 
 const router = express.Router();
+const userController = new UserController();
 
 router.use(cors());
 router.use(bodyParser.json());
 
 router.get('/hello', helloController.get);
+router.post('/users', validateUser, (req, res) => {
```

Comment:
@marcianszlovak this is a nice idea, but the user should not be saved when it is not validated, so the `validateUser` is not a middleware, but it's part of the userService/userRepo where the actual save happens

Reply:
I'm not sure what you mean, the validator prevents invalid users from registering. You can see it in the route test file even, when I mock registering an invalid user.

![post api-users](https://user-images.githubusercontent.com/56042419/87063656-ed9c5e00-c20e-11ea-8350-31352d5f8c10.png)

Reply:
I understand that, but the user should not be saved from anywhere if it's not valid :)
right now if I create another endpoint let's say where an admin can create a user it won't go through the same checks, unless I specifically use the validator there again. I as a developer should not be able to save a user without going through the validation